### ISSUE TITLE
Video Ripper Broken

### ISSUE THREAD

**Issue by rephormat** [4pr0n/ripme#549]
No longer able to download videos.  Sample error log from xHamster is attached.

Affected versions: >= 1.4.14

[ripme.txt](https://github.com/4pr0n/ripme/files/1014765/ripme.txt)

**Comment by cyian-1756** [comment]
Can you give an example link?

**Comment by cyian-1756** [comment]
After a bit of looking it seems that one of the updates since 1.4.2 has broken the ripper, I'll diff the versions and see if I can fix it

**Comment by cyian-1756** [comment]
It looks like it was some of the changes @metaprime made [here](https://github.com/4pr0n/ripme/compare/1.4.13...1.4.14#diff-afd0b40c3fb93545950ba5fbb83ce413) that broke the video ripper. 

Replacing that file with the version from 1.4.13 fixes the error

**Comment by rephormat** [comment]
Confirmed @cyian-1756 findings.  The video ripper is broken for all supported sites.  Updating Issue.

**Comment by metaprime** [comment]
@cyian-1756 which file did you replace to fix it?

**Comment by metaprime** [comment]
Oh it's this change (part of #185):

```
-            } catch (Exception e) {
+            } catch (InstantiationException e) {
+                // Incompatible rippers *will* throw exceptions during instantiation.
+            } catch (IllegalAccessException e) {
+                // Incompatible rippers *will* throw exceptions during instantiation.
+            } catch (IllegalArgumentException e) {
+                // Incompatible rippers *will* throw exceptions during instantiation.
+            } catch (InvocationTargetException e) {
                 // Incompatible rippers *will* throw exceptions during instantiation.
             }
```

I wasn't a fan of this to begin with (but no strong opinion so I merged it anyway). There's no value in enumerating each type of exception when the handling is the same. And look, it introduced a bug by not handling all cases.

This also points out the need for improving test coverage and fixing up the CI...

**Comment by metaprime** [comment]
Fixed in 1.4.18
